### DISCUSSION TITLE
Different statistics P value in ggpubr and Findmarkers()

### DISCUSSION THREAD

**Discussion by Bouncysig18** [satijalab/seurat#6811] [Q&A]
Hello, everyone!

i use seurat to analyze scRNA-seq data, i got a interested "cluter46". when i used ` FindMarkers(c46.intergrated,group.by = "orig.ident",logfc.threshold = 0.1,ident.1 = "KO", ident.2="WT",test.use="wilcox")`. i got a DEG, GeneA, with significant p value and padj value **(<0.05, Significant GeneA difference between "KO" and "WT").** But when i use `Vlnplot(c46.intergrated,group.by = "orig.ident",features="GeneA",)+stat_compare_means(comparisons = comparision_list,label =  "p.signif",method = "wilcox")`. I got GeneA with non-significant p value and padj value **(>0.05, No significant GeneA difference between "KO" and "WT").**

**i have read the "Statistics in violin plots #5997"**. i knew that samuel-marsh recommanded us to use the p value or p adj value of FindMarkers. But i'm confused **why there are some differences bewtten same statistics anaylsis (wilcox) and what'is the difference between two statistics calculations.**

Can you give me answers about this question? 

Bests,
Wenhao

**Comment by Bouncysig18** [comment]
@samuel-marsh Could you help me? please

**Reply by samuel-marsh** [reply]
Hi,
So they are doing very different things because of how test and multiple comparisons are working in the object vs only on the data used for a specific plot.  The short answer is that ggpubr on the data used just for the plot is not what you want to do.  It’s not performing multiple comparisons to the whole dataset and so shouldn’t be used in place of true DE analysis.

Depending on the situation cell level DE (e.g. FindMarkers) may not be appropriate either but that’s different topic (cluster v cluster DE and appropriate accounting for replicates and other variables, etc).

Best,
Sam

**Reply by Bouncysig18** [reply]
> Hi, So they are doing very different things because of how test and multiple comparisons are working in the object vs only on the data used for a specific plot. The short answer is that ggpubr on the data used just for the plot is not what you want to do. It’s not performing multiple comparisons to the whole dataset and so shouldn’t be used in place of true DE analysis.
> 
> Depending on the situation cell level DE (e.g. FindMarkers) may not be appropriate either but that’s different topic (cluster v cluster DE and appropriate accounting for replicates and other variables, etc).
> 
> Best, Sam

Hi, Sam

Thanks for your @samuel-marsh  your  replying. My situation is that i want to compare the difference between the expression of one gene, "GeneA" of two treatments. In my view, this situtation don't need multiple comparisons (is it right ?).

But FindMarkers (like this FindMarkers(c46.intergrated,group.by = "orig.ident",logfc.threshold = 0.1,ident.1 = "KO", ident.2="WT",test.use="wilcox")) will do multiple comparisons (is it right ? and are only two group necessary to do multiple comparisons ?)

And i cannot understand why the Pvalue of FindMarkers (cell level DE) is not appropriate to my situation. Beacause in my view, this pvalue in FindMarkers results is to tell us how significant the expression difference of marker genes are. if not appropriate indeed for my sitution, can you recommand me for an appropriate statistical test way?

Happy new years,
Wenhao

**Reply by samuel-marsh** [reply]
Hi,

So this is fundamental aspect of differential expression for RNA-seq analysis.  There are bunch of resources you can find on web that go into it in more detail.  One example here: https://hbctraining.github.io/DGE_workshop/lessons/05_DGE_DESeq2_analysis2.html (See section "Multiple test correction").

In terms of whether cell-level DE is appropriate that depends on test (and whether chosen test is accounting for other variables; i.e. batch, biological replicates, etc).  If you just run wilcoxon rank sum test in FindMarkers you cannot provide any other variables to the latent variables parameter and are potential creating issues by not taking into account sample level differences (if one sample in group has drastic change so much that it swings DE on cell level but that change is not replicated across biological replicates then that could be missed).   Mixed models (e.g. MAST) perform relatively well when providing appropriate variables to the fixed and random effects in the model (Suggest running directly through MAST package as Seurat implementation of MAST does not allow for providing variables to the random effect in the model).

Outside of mixed models several papers have highlighted the power and utility of pseudobulk methods when performing sample-level comparisons between groups in scRNA-seq data.  Two examples:
https://www.nature.com/articles/s41467-020-19894-4
https://www.nature.com/articles/s41467-022-35519-4

Best,
Sam
